lib/route-helper.js

```diff
@@ -80,136 +80,94 @@ var routeHelper = module.exports = {
+    var returns = routeHelper.convertReturnsToSwagger(route, typeRegistry);
+    var defaultCode = route.returns && route.returns.length ? 200 : 204;
+    // TODO - support strong-remoting's option for a custom response code
+
+    var responseMessages = {};
+    responseMessages[defaultCode] = {
+      description: 'Request was successful',
+      schema: returns,
+      // TODO - headers, examples
+    };
+
     if (route.errors) {
-      responseMessages.push.apply(responseMessages, route.errors);
+      // TODO define new LDL syntax that is status-code-indexed
+      // and which allow users to specify headers & examples
+      route.errors.forEach(function(msg) {
+        responseMessages[msg.code] = {
+          description: msg.message,
+          schema: schemaBuilder.buildFromLoopBackType(msg.responseModel,
+                                                      typeRegistry),
+          // TODO - headers, examples
+        };
+      });
     }
 
     debug('route %j', route);
 
-    var responseDoc = modelHelper.LDLPropToSwaggerDataType(returns);
+    var tags = [];
+    if (classDef && classDef.name) {
+      tags.push(classDef.name);
+    }
 
-    var apiDoc = {
+    var entry = {
       path: routeHelper.convertPathFragments(route.path),
-      // Create the operation doc.
-      // We are using extendWithType to use `type` for the top-level (200)
-      // response type. We use responseModels for error responses.
-      // see https://github.com/strongloop/loopback-explorer/issues/75
-      operations: [routeHelper.extendWithType({
-        method: routeHelper.convertVerb(route.verb),
-        // [strml] remove leading model name from op, swagger uses leading
-        // path as class name so it remains unique between models.
-        // route.method is always #{className}.#{methodName}
-        nickname: route.method.replace(/.*?\./, ''),
-        deprecated: route.deprecated,
-        consumes: ['application/json', 'application/xml', 'text/xml'],
-        produces: ['application/json', 'application/javascript', 'application/xml', 'text/javascript', 'text/xml'],
-        parameters: accepts,
-        responseMessages: responseMessages,
-        type: returns.model || returns.type || 'void',
+      method: routeHelper.convertVerb(route.verb),
+      operation: {
+        tags: tags,
         summary: typeConverter.convertText(route.description),
-        notes: typeConverter.convertText(route.notes)
-      }, returns)]
+        description: typeConverter.convertText(route.notes),
+        // [bajtos] We used to remove leading model name from the operation
+        // name for Swagger Spec 1.2. Swagger Spec 2.0 requires
+        // operation ids to be unique, thus we have to include the model name.
+        operationId: route.method,
+        // [bajtos] we are omitting consumes and produces, as they are same
+        // for all methods and they are already specified in top-level fields
```

Comment:
I know. What do you propose? Should we detect whether consumes/produces differs from the default and include it here in such case? I am not sure if strong-remoting actually supports per-method customization of consumes/produces config. Should I just rephrase the comment?